text/trace/0192-messaging-semantic-conventions-spec.md

```diff
@@ -0,0 +1,628 @@
+auto-generated identifier (like a conversation id or an auto-generated name for
+an anonymous destination).
+
+> The span name SHOULD consist of the name of the messaging system followed by
+> an [operation name](#operation-name). The destination name MAY be appended if
+> it is of low cardinality.
+
+##### Examples
+
+* `kafka publish shop.orders`
+* `rabbitmq receive print_jobs`
+* `AmazonSQS deliver`
+* `activemq settle`
+
+#### Operation name
+
+The following operations related to messages are covered by these semantic
+conventions:
+
+| Operation name | Description |
+|----------------|-------------|
+| `publish`      | One ore more messages are provided for publishing to an intermediary. |
+| `create`       | A message is created. |
+| `receive`      | One or more messages are requested by a consumer. |
+| `deliver`      | One or more message are passed to a consumer. |
+| `settle`       | One or more message are settled. |
+
+For further details about each of those operations refer, to the [section about trace structure](#trace-structure).
+
+#### Span kind
+
+[Span kinds](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#spankind)
+SHOULD be set according to the following table, based on the operation a span describes.
+
+| Operation name | Span kind|
+|----------------|-------------|
+| `publish`      | `PRODUCER`, if no `create` spans are present. `INTERNAL` otherwise. |
+| `create`       | `PRODUCER` |
+| `receive`      | `CONSUMER` |
+| `deliver`      | `CONSUMER` |
+| `settle`       | `INTERNAL` |
+
+Setting span kinds according to this table ensures, that span links between
+consumer and producers always go from a `PRODUCER` span on the producer side to
+a `CONSUMER` span on the consumer side. This allows analysis tools to interpret
+linked traces without the need of additional semantic hints.
+
+#### Attributes
+
+Attribute             | Type   | Required
+----------------------|--------|---------
+[`messaging.broker`](#messagingbroker) | string | Yes
+[`messaging.operation`](#messagingoperation) | string | Yes
+[`messaging.destination.name`](#messagingdestinationname) | string | For producer spans
+[`messaging.destination.template`](#messagingdestinationtemplate) | string | No
+[`messaging.destination.kind`](#messagingdestinationkind) | string | No
+[`messaging.destination.temporary`](#messagingdestinationtemporary) | string | No
+[`messaging.destination.anonymous`](#messagingdestinationanonymous) | string | No
+[`messaging.source.name`](#messagingsourcename) | string | For consumer spans
+[`messaging.source.template`](#messagingsourcetemplate) | string | No
+[`messaging.source.kind`](#messagingsourcekind) | string | No
+[`messaging.source.temporary`](#messagingsourcetemporary) | string | No
+[`messaging.source.anonymous`](#messagingsourceanonymous) | string | No
+[`messaging.protocol.name`](#messagingprotocolname) | string | No
```

Comment:
we should figure out the relationship with  `net.transport` attribute. Also, if we're not targeting transport-level instrumentation in v1, should we really add protocol information here? 
Ideally, when transport-level instrumentation is added, all the information will be there